client/extensions/zoninator/components/search-autocomplete/suggestions.jsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { find, findIndex, first, isEqual, map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import QueryPosts from 'components/data/query-posts';
+import { getSitePostsForQuery } from 'state/posts/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import SuggestionItem from './suggestion-item';
+
+class Suggestions extends Component {
+
+	static propTypes = {
+		suggest: PropTypes.func.isRequired,
+		searchTerm: PropTypes.string,
+		suggestions: PropTypes.array,
+		ignored: PropTypes.array,
+		siteId: PropTypes.number,
+	}
+
+	static defaultProps = {
+		suggestions: [],
+		searchTerm: '',
+	}
+
+	state = {
+		suggestions: [],
+		suggestionPosition: 0,
+		currentSuggestion: null,
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if (
+			isEqual( nextProps.suggestions, this.props.suggestions ) &&
+			isEqual( nextProps.ignored, this.ignored )
+		) {
+			return;
+		}
+
+		const suggestions = this.filterSuggestions( nextProps.suggestions, nextProps.ignored );
+
+		this.setState( {
+			suggestions,
+			suggestionPosition: 0,
+			currentSuggestion: first( suggestions ),
+		} );
+	}
+
+	filterSuggestions( suggestions, ignored ) {
+		if ( ! suggestions ) {
+			return [];
+		}
+
+		return suggestions.filter( ( { slug } ) => ! find( ignored, { slug } ) );
+	}
+
+	countSuggestions = () => this.state.suggestions.length;
+
+	getPositionForSuggestion = slug => findIndex( this.state.suggestions, { slug: slug } );
+
+	getSuggestionForPosition = position => this.state.suggestions[ position ];
+
+	increasePosition() {
+		const position = ( this.state.suggestionPosition + 1 ) % this.countSuggestions();
+		this.setState( {
+			suggestionPosition: position,
+			currentSuggestion: this.getSuggestionForPosition( position ),
```

Comment:
`currentSuggestion` is kinda duplicating information. Maybe only store `suggestionPosition` in state, and only use `getSuggestionForPosition()` when needing to obtain (read) the current suggestion (but don't set it as part of state)?

Reply:
Maybe combine `this.props.suggest()` and `getSuggestionForPosition()` into a new `this.suggest()` method?